app/models/rook.rb

```diff
@@ -2,4 +2,11 @@ class Rook < Piece
   def to_unicode
     black? ? "\u265C" : "\u2656"
   end
+
+  def valid_move?(destination_x, destination_y)
+    distance_x = (destination_x - x_position).abs
+    distance_y = (destination_y - y_position).abs
+
+    distance_x == distance_y || distance_x.zero? || distance_y.zero?
```

Comment:
This should be `distance_x.zero? || distance_y.zero?` because what you currently have is the queen's valid move method
Also you need to add this in factories.rb ```factory :rook do
    association :game
  end``` this should solve your rspec failing problem